File: Data Structures & Algorithms/jump-game-ii/submission-4.cs
public class Solution {
    public int Jump(int[] nums) {
        return Dfs(nums, 0);
    }

    private int Dfs(int[] nums, int i) {
        if (i == nums.Length - 1) {
            return 0;
        }
        if (nums[i] == 0) return 1000000;
        int res = 1000000;
        int end = Math.Min(nums.Length - 1, i + nums[i]);
        for (int j = i + 1; j <= end; j++) {
            res = Math.Min(res, 1 + Dfs(nums, j));
        }
        return res;
    }
}

File: Data Structures & Algorithms/coin-change/submission-4.js
class Solution {
    /**
     * @param {number[]} coins
     * @param {number} amount
     * @return {number}
     */
    coinChange(coins, amount) {

        const dfs = (amount) => {
            if (amount === 0) return 0;

            let res = Infinity;
            for (let coin of coins) {
                if (amount - coin >= 0) {
                    res = Math.min(res, 
                          1 + dfs(amount - coin));
                }
            }
            return res;
        }

        const minCoins = dfs(amount);
        return minCoins === Infinity ? -1 : minCoins;
    }
}